File: JavaScript/Searching/LinearSearch.js
/*
Linear search also known as sequential search is an iterative search technique for finding an element within a list. It sequentially checks each element of the list until a match is found or the whole list has been searched.
*/


const prompt = require("prompt-sync")({ sigint: true });


const linearSearch = (arr, key) => {
    // Traversing through the array elements
    for (let i = 0; i < arr.length; i++)
    {
        // Checking if the current element is equal to the number to be found
        if (arr[i] === key)
        {
            return true;
        }
    }
    return false;
}

// Taking input from the user
let n = +prompt("Enter the number of elements : ");

let array = [];
for (let i = 1; i <= n; i++) {
  array.push(+prompt("Enter value : "));
}

const key = +prompt("Enter number to search : ");

// Function call to check the presence of the number
if (linearSearch(array, key))
{
    console.log("Element found in array!");
}
else 
{
    console.log("Element not found in array!");
}

/*

TIME COMPLEXITY: O(n), where n is the size of the array

SPACE COMPLEXITY: O(1)

-->SAMPLE I/O 1:
Enter the number of elements : 5
Enter value - 15
Enter value - 1
Enter value - 67
Enter value - 23
Enter value - 66
Enter number to search : 6
Element not found in array!

-->SAMPLE I/O 1:
Enter the number of elements : 5
Enter value : 15
Enter value : 2
Enter value : 69
Enter value : 4
Enter value : 18
Enter number to search : 18
Element found in array!

*/

File: C++/Searching/LinearSearch.cpp
/*
Linear search also known as sequential search is an iterative search technique for finding an element within a list. It sequentially checks each element of the list until a match is found or the whole list has been searched.
*/


#include <iostream>

using namespace std;

bool linearSearch(int arr[], int n, int key)
{
    // Traversing through the array's each element
    for (int i = 0; i < n; i++)
    {
        // Checking if the element is equal to the 
        // the key element (to be searched)
        if (arr[i] == key)
        {
            return true;
        }
    }

    // Search failure
    return false;    
}[


int main() 
{
    // Driver code to get user input and generate output
    int n;
    cout<<"\nEnter number of elements in array: ";
    cin>>n;

    int arr[n];
    int key;
    for (int i = 0; i < n; i++)
    {
        cout<<"\nEnter value for element "<<i+1<<": ";
        cin>>arr[i];
    }

    cout<<"\nEnter element to be searched: ";
    cin>>key;

    if (linearSearch(arr, n, key))
    {
        cout<<"\nThe element was found!\n";
    }
    else
    {
        cout<<"\nThe element is not present in the given array!\n";
    }

    return 0;

}
/*

TIME COMPLEXITY: O(n) where n = size of array

SPACE COMPLEXITY: O(1)

-- > SAMPLE I/O 1:

Enter number of elements in array: 6

Enter value for element 1: 12

Enter value for element 2: 90

Enter value for element 3: 0

Enter value for element 4: 10

Enter value for element 5: 45

Enter value for element 6: 31

Enter element to be searched: 90

The element was found!

-- > SAMPLE I/O 2:

Enter number of elements in array: 5

Enter value for element 1: 12

Enter value for element 2: 34

Enter value for element 3: 90

Enter value for element 4: 78

Enter value for element 5: 51

Enter element to be searched: 8

The element is not present in the given array!

*/

